src/server/game/Entities/Player/Player.cpp

```diff
@@ -2217,6 +2217,16 @@ void Player::RemoveFromGroup(Group* group, ObjectGuid guid, RemoveMethod method
     group->RemoveMember(guid, method, kicker, reason);
 }
 
+uint32 Player::GetXP() const
+{
+    return m_activePlayerData->XP;
+}
+
+uint32 Player::GetXPForNextLevel() const
+{
+    return m_activePlayerData->NextLevelXP;
+}
+
```

Comment:
These getters can be directly in the .h file.

Reply:
I usually only use getters in .h files if the fields are structs/primitives (not pointers).
But sure, I'll do that.

Reply:
If you look at `m_activePlayerData` getters (or any simple "UF" getters really) they're all in .h 😄